packages/vm/examples/7702/uniswap-swap-transfer.ts

```diff
@@ -0,0 +1,107 @@
+  const privateKey = hexToBytes(privateKeyHex)
+  const userAddress = createAddressFromPrivateKey(privateKey)
+  console.log('EOA:', userAddress.toString())
+
+  // ─── set up EthereumJS VM with EIP-7702 enabled ───────────────────────
+  const common = new Common({
+    chain: Mainnet,
+    hardfork: Hardfork.Cancun,
+    eips: [7702],
+  })
+  const stateManager = new RPCStateManager({
+    provider: 'YourProviderURLHere',
+    blockTag: 22_000_000n,
+  })
+  const vm = await createVM({ common, stateManager })
+
+  // ─── constants & ABIs ────────────────────────────────────────────────
+  const DAI = '0x6B175474E89094C44Da98b954EedeAC495271d0F'
+  const UNISWAP_V3_ROUTER = '0xE592427A0AEce92De3Edee1F18E0157C05861564'
+  const WETH = '0xC02aaa39b223FE8D0A0e5c4F27EaD9083C756Cc2'
+  const COLD_WALLET = `0x${'42'.repeat(20)}`
+  const BATCH_CONTRACT = '0xYourBatchContractAddressHere'
+
+  const erc20Abi = [
+    'function approve(address _spender, uint256 _amount) external returns (bool success)',
+    'function transfer(address _to, uint256 _value) public returns (bool success)',
+  ]
+  const routerAbi = ['function exactInput(bytes)']
+  const batchAbi = ['function executeBatch(bytes[] calldata) external']
+
+  const erc20 = new Interface(erc20Abi)
+  const router = new Interface(routerAbi)
+  const batchContract = new Interface(batchAbi)
+
+  // ─── trade parameters ────────────────────────────────────────────────
+  const amountIn = parseUnits('10000', 18) // 10000 DAI
+  const amountOut = parseEther('4') // expect at least 4 WETH
+
+  // ─── encode your underlying swap data ───────────────────────────────────
+  const uniswapV3SwapPayload = `0xYourSwapCallDataHere`
+
+  // ─── encode your three sub-calls ───────────────────────────────────────
+  // 1) DAI approve
+  const callApprove = erc20.encodeFunctionData('approve', [
+    UNISWAP_V3_ROUTER,
+    amountIn,
+  ]) as PrefixedHexString
+
+  // 2) Uniswap V3 swapExactInput
+  const callSwap = router.encodeFunctionData('exactInput', [
+    uniswapV3SwapPayload,
+  ]) as PrefixedHexString
+
+  // 3) sweep WETH to cold wallet
+  const callTransfer = erc20.encodeFunctionData('transfer', [
+    COLD_WALLET,
+    amountOut,
+  ]) as PrefixedHexString
+
+  const calls = [callApprove, callSwap, callTransfer].map(hexToBytes)
+
+  // ─── sign authorization for each ──────
+  const targets: PrefixedHexString[] = [DAI, UNISWAP_V3_ROUTER, WETH]
+  const auths = targets.map((address, i) =>
+    eoaCode7702SignAuthorization({ chainId: '0x1', address, nonce: `0x${i}` }, privateKey),
+  )
+
+  // ─── build & send your single 7702 tx ───────────────────────────────
+  const batchData = batchContract.encodeFunctionData('executeBatch', [calls]) as `0x${string}`
+
+  const txData: EOACode7702TxData = {
+    nonce: 0n,
+    gasLimit: 1_000_000n,
+    maxFeePerGas: parseUnits('10', 9), // 10 gwei
+    maxPriorityFeePerGas: parseUnits('5', 9), // 5 gwei
+    to: BATCH_CONTRACT,
```

Comment:
This will call the batch contract. It will therfore interact with WETH/DAI/Uniswap from the batch contract. We want to interact with it from our EOA (which is why EOA should be delegated to BATCH_CONTRACT)